device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSenderLinkHandler.java

```diff
@@ -36,10 +37,16 @@
     private long nextTag = 0;
     private final AmqpsLinkStateCallback amqpsLinkStateCallback;
 
-    AmqpsSenderLinkHandler(Sender sender, AmqpsLinkStateCallback amqpsLinkStateCallback, String linkCorrelationId)
+    AmqpsSenderLinkHandler(Sender sender, AmqpsLinkStateCallback amqpsLinkStateCallback, String linkCorrelationId, String modelId)
     {
         this.amqpProperties = new HashMap<>();
         this.amqpProperties.put(Symbol.getSymbol(API_VERSION_KEY), TransportUtils.IOTHUB_API_VERSION);
+
+        if (modelId != null && !modelId.isEmpty())
+        {
+            this.amqpProperties.put(Symbol.getSymbol(PNP_MODEL_ID_KEY), modelId);
```

Comment:
Do we need to worry about URL escaping the model Id for AMQP? I forget. I know we had to for MQTT.

Reply:
Looks like we don't need to. We URL escaped the model Id in MQTT because it was embedded into a URL. Here in AMQP though, it is just the value in a key-value pair. I checked, and we are not URL escaping the model Id for AMQP in the .NET SDK either.